vignettes/stages.Rmd

```diff
@@ -0,0 +1,97 @@
+A function is considered stable when the author is happy with its interface, doesn't see major issues, and is happy to share it with the world.
+Because this stage is the default, functions will only be given a stable badge if there's some specific need to draw attention to their status.
+
+Stability is defined in terms of breaking changes.
+A **breaking change** is a change that breaks code that uses the function as expected.
+In general, breaking changes reduce the set of code that works without error, either by removing a function, removing a function argument, or decreasing the set of valid inputs.
+Breaking changes also include changes to output type.
+If you imagine all the possible inputs to a function that return a result (and not an error), a breaking change makes the set smaller.
+
+Not all changes that cause your to stop working are breaking changes.
+For example, you might have accidentally relied on a bug.
+When the bug is fixed, your code breaks, but this is not a breaking change.
+
+Stable functions come with two promises related to breaking changes:
+
+-   Breaking changes will be avoided where possible.
+    We'll only make breaking changes if we consider the long term benefit of the change to be greater than short term pain of changing existing code.
+
+-   If a breaking change is needed, it will occur gradually, through the deprecation process described next.
+    This gives you plenty of time to adjust your code before it starts generating errors.
+
+## Deprecated
+
+A ![Deprecated](figures/lifecycle-deprecated.svg){style="vertical-align:middle"} function has a better alternative available and is scheduled for removal.
+When you call a deprecated function, you get a warning telling you what to use instead.
+For example, take `tibble::as_data_frame()`:
+
+```{r, eval = FALSE}
+df <- tibble::as_data_frame(x = 1)
+#> Warning message:
+#> `data_frame()` is deprecated as of tibble 1.1.0.
+#> Please use `tibble()` instead.
+#> This warning is displayed once every 8 hours.
+#> Call `lifecycle::last_warnings()` to see where this warning was generated. 
+```
+
+The deprecating warning tells you when the function was deprecated (in tibble 1.1.0 released in 2016), and what to use instead (`tibble()`).
+To avoid being too annoying, deprecation messages will only appear once per session, and you can find out exactly where they come from by calling `lifecycle::last_warning()`.
+`vignette("user")` provides more advice on handling deprecation warnings in your code.
+
+Particularly important functions may go through two additional stages of deprecation:
+
+-   **Soft deprecated** comes before deprecated.
+    It's a gentler form of deprecation designed to prevent new uses of a function and encourage package developers to move away from it.
+    Soft deprecated allows a package to change its interface to encourage package developers to update their code before users are forced to change.
```

Comment:
```suggestion
    Soft deprecated allows a package to change its interface to encourage package developers to update their code before their users are forced to change.
```

Also mention that users in the global workspace also get a warning? Only indirect uses don't warn.

Reply:
I think that's covered by the previous sentence, and you can get more details from the docs.